runtime/arm64.S

```diff
@@ -898,11 +1216,14 @@ L(frame_runstack):
         ldr     TMP, Caml_state(c_stack)
         mov     sp, TMP
         bl      G(caml_free_stack)
-    /* switch directly to parent stack with correct return */
-        mov     x0, x20
-        mov     x1, x19
+    /* switch directly to parent stack */
         mov     sp, x21
         CFI_RESTORE_STATE
+    /* Signal to TSan that we exit caml_runstack (no registers to save here) */
+        TSAN_EXIT_FUNCTION
+    /* pick correct return value */
+        mov     x0, x20
+        mov     x1, x19
```

Comment:
Does the re-ordering here of moving the return value loading impact the CFI_RESTORE_STATE directive? It seems correct with respect to the TSAN_EXIT_FUNCTION behaviour here and in `amd64.S`

Reply:
No problem here. The `CFI_RESTORE_STATE` a few lines above matches the `CFI_REMEMBER_STATE` a few more line above, then `TSAN_EXIT_FUNCTION` keeps the stack state unchanged upon return, and reloading the possible return values is done from registers, so no stack impact either.